### PR TITLE
osf-credentials command for credential input and verification

### PR DESCRIPTION
**Pull request by mih**
Closes gh-92

### CODE REVIEW

**Comment by adswa** [comment]
Would it make sense to mimic the provider credential management behavior and ask whether the user wants to reconfigure their credentials if a 403 unauthorized response is returned?

**Comment by mih** [comment]
> Would it make sense to mimic the provider credential management behavior and ask whether the user wants to reconfigure their credentials if a 403 unauthorized response is returned?

I'm not sure what you mean. Within `osf-credentials`? Or whenever something fails elsewhere?

ATM, if the credentials are found to be invalid, they are deleted from the DB and one can restart the command to enter new ones.

**Comment by adswa** [comment]
Oh, then I might be doing something wrong?
```
datalad create-sibling-osf -s osf myosf
You need to authenticate with 'https://osf.io' credentials. https://osf.io/settings/tokens provides information on how to gain access
token:  # was too lazy to find it
You need to authenticate with 'https://osf.io' credentials. https://osf.io/settings/tokens provides information on how to gain access
token (repeat): 
You need to authenticate with 'https://osf.io' credentials. https://osf.io/settings/account provides information on how to gain access
user: adina.wagner@t-online.de

password: 
password (repeat): 
# first unauthorized error
[ERROR  ] 401 Client Error: Unauthorized for url: https://api.osf.io/v2//nodes/ [models.py:raise_for_status:941] (HTTPError) 
```
```
# second attempt does not ask for credentials again
(hbmhack) ╭─adina@muninn /tmp/blabla on master
╰─➤ datalad create-sibling-osf -s osf myosf                                 1 ↵
[ERROR  ] 401 Client Error: Unauthorized for url: https://api.osf.io/v2//nodes/ [models.py:raise_for_status:941] (HTTPError) 
```

**Comment by mih** [comment]
Ah, the idea is that we instruct people to run `datalad osf-credentials` once before using any command. In this case, none of that would (should) happen.

**Comment by adswa** [comment]
Ah, thanks for clarifying! :facepalm: 

The result renderer reports ``None`` as this is the (non-existing) path, right? 
```
osf_credentials(ok): None [authenticated as Adina Svenja Wagner <adina.wagner@t-online.de>]
```

File: datalad_osf/credentials.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        if reset and auth.is_known:
+            auth.delete()
+        cred = {v: auth().get(k, None) for k, v in cred_spec.items()}
+
+        # now verify that the credentials work by querying the
+        # logged in user
+        osf = OSF(**cred)
+        try:
+            req = osf.session.get('https://api.osf.io/v2/users/me/')
+            if req.status_code != 200:
```
Comment (bpoldrack) [inline datalad_osf/credentials.py:91]: IIRC this kinda should be `req.raise_for_status()`.

**Comment by mih** [comment]
> Ah, thanks for clarifying! 
> 
> The result renderer reports `None` as this is the (non-existing) path, right?
> 
> ```
> osf_credentials(ok): None [authenticated as Adina Svenja Wagner <adina.wagner@t-online.de>]
> ```

Yes, I will prepare a PR for datalad-core to deal with that.

File: datalad_osf/credentials.py (hunk 1 above)
Comment (mih) [reply datalad_osf/credentials.py:91]: True! Thx!

**Comment by mih** [comment]
> The result renderer reports `None` as this is the (non-existing) path, right?

@adswa https://github.com/datalad/datalad/pull/4666

**Pull request merged**